lib/src/barback/dart2js_transformer.dart

```diff
@@ -293,8 +287,47 @@ class _BarbackCompilerProvider implements dart.CompilerProvider {
     });
   }
 
-  /// A [CompilerOutputProvider] for dart2js.
-  EventSink<String> provideOutput(String name, String extension) {
+  Future<String> _readResource(Uri url) {
+    return new Future.sync(() {
```

Comment:
Can you convert this to `async`/`await`?

Reply:
I thought I could get away with still using `compiler_unsupported` v1.22 but there have been changes to the new API since then. I will land a new version of `compiler_unsupported` then update this change.

The diagnostic handler is still a copy of `FormattingDiagnosticHandler`

Reply:
Done